src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -4132,7 +4132,8 @@ trait Types
       while (tparams1 != tparams0) {
         tparams0 = tparams1
         tparams1 = tparams filter { p =>
-          tparams1 exists { p1 => p1 == p || (p1.info contains p) }
+          tparams1.exists( (p1: Symbol) => p1 eq p) ||
```

Comment:
I wonder if it would make sense to use a Set of Symbols and a Set of Types (`p1.map(_.info)`)? How often are the `p1.info` the same and we keep testing against them?

Reply:
What I am trying to do is to have another "border" collection, that in each iteration keeps the elements found in the latest iteration, and run the exists only on that. This would avoid repeating calls `p1.info.contains(p)` twice for the same pair of `p1, p`.